CONTRIBUTING.rst

```diff
@@ -0,0 +1,76 @@
+
+A contribution can be either be an **patch**, **feature**, or **hotfix**:
+ * **Patch** include bugfixes and minor changes to the code and goes in Patch releases (usually made on a bi-weekly schedule).
+ * **Feature** include major developments or potentially disruptive changes and goes in Feature release (made multiple times a year).
+ * **Hotfix** are specific patch happening due to the necessity of important fixes and goes in postfix release (made when needed).
+
+Accordingly, the `repository <https://github.com/rucio/rucio/>`_  consists of three different branches:
+ * the **master** branch includes the patch/minor development of the current major version.
+ * the **next** branch includes the development for the next major version.
+ * the **hotfix** branch includes the patch for postfix releases.
+
+Thus, on release day of a feature release both master and next are the same,
+afterwards they diverge until the next feature release.
+Pull requests for **features** are only made against the **next** branch.
+Pull requests for **patches** are made against the **next** and **master** branch.
+Simultaneously, as these bugfixes need to be represented in both branches. Thus two
+pull requests are needed for patches, and the helper scripts do it
+automatically for you.
+
+Setting up the repository
+=========================
+
+**Step 1**: Fork the `repository <https://github.com/rucio/rucio/>`_ on Github
+
+**Step 2**: Clone the repository to your development machine and configure it::
+
+  $ git clone https://github.com/<YOUR_USER>/rucio/
+  $ git remote add YOUR_USER https://github.com/YOUR_USER/rucio.git
+  $ git remote add upstream https://github.com/rucio/rucio.git
+  $ git branch --track next
+
+We can either establishing a virtualenv_. to run rucio within. A docker image is also
+doable (cf. developing notes)
+
+.. _virtualenv: http://www.virtualenv.org/
+
+
+Contributing
+============
+
+**Step 1**: Create an `issue <https://github.com/rucio/rucio/issues/new>`_ with the description
+of the contribution (motivation, modification and expected results) with the
+label **Patch**, **Feature** or **Hotfix** and a milestone. Every issue will
+get a **unique issue number**.
+
+**Step 2**: Create a local branch that corresponds to the issue. There are utility scripts to help you with this::
+
+  $ ./tools/create-patch-branch <unique issue number> '<component> #<issue number>: <short_change_message>'
+  $ ./tools/create-feature-branch <unique issue number> '<component> #<issue number>: <short_change_message>'
+
+**Step 3**: Commit your change. The format of the commit message must be::
+
+<component>: <change_message> Fix #<issue number>
+
+Valid component names are listed in the `label list <https://github.com/rucio/rucio/labels>`_
+
+**Step 4**: Push the commit to your forked repository and create the pull request(s). There is a helper script to assist you::
+
+  $ ./tools/submit-pull-request
+
+**Step 5**: Watch the pull request for comments
+
```

Comment:
I think the text reads well like this but I still think it is a good idea that the review process specified in the contribution guide. So one more section:

******
Review
******

- All active Rucio developers are encouraged to comment pull requests.
- All collaborators, thus the Rucio core development team, can approve, deny, or request changes for pull requests.
- Pull requests are merged by the Rucio development lead.

Reply:
more something among the collaborators/rucio core dev. It's not for the contributors guides (short and concise). This can go in the dev. section of the docs (imo)

Reply:
I think it's good to have it here, most other contributing guides I have seen have this too. It's only 3 lines anyway and people know what to expect.

Reply:
Would you have examples ?
approve, deny or request are actually features of the pull request and review process so it's kind of obvious (imo). Encouragement we always did and merge it's 99% you.  I was more thinking of having the collaborator part better described elsewhere. Actually where is done the mapping component <-> person ? to have suggested people for review ?

Reply:
Pretty much all the big open source projects describe their review process, CMake, sphinx, mongo etc. I think at some point, when we have a more elaborate guide we can take many parts out of this document and just link the different chapters or something, but for the moment that's what we have and I think it should be there.
The mapping component <-> person I was at one point thinking maybe to the wiki, but would be strange to activate this function just for this feature. So I guess readthedocs a page about it?

Reply:
when a pull request is done, sometimes github proposes reviewers. I'm not sure it has to be done manually or it's based on stats of the people modifying the code

Reply:
Yes, but for now it is not super correct, lets see what the github ML brings us in the future 😄